hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java

```diff
@@ -442,7 +442,7 @@ protected ProcedureMetrics getProcedureMetrics(MasterProcedureEnv env) {
   private boolean prepareMergeRegion(final MasterProcedureEnv env) throws IOException {
     // Fail if we are taking snapshot for the given table
     TableName tn = regionsToMerge[0].getTable();
-    if (env.getMasterServices().getSnapshotManager().isTakingSnapshot(tn)) {
+    if (env.getMasterServices().getSnapshotManager().isTableTakingAnySnapshot(tn)) {
```

Comment:
+1

SplitTableRegionProcedure has the same problem. Maybe we can fix it also ? Or File a new issue ?

Reply:
Let's change the title of the issue and then fix them both.

And also better include the UT provided by @frostruan ?

Reply:
Alright, that's a great suggestion, I'll add it now.